client/www/scripts/modules/manager/manager.controllers.js

```diff
@@ -609,6 +616,59 @@ Manager.controller('ManagerMainController', [
       }
     };
 
+    function updateHostEnv(host, env) {
+      var deferred = $q.defer();
+      var command =  {
+        cmd: 'env-set',
+        env: env
+      };
+
+      if (host) {
+        host.action(command, function(err, res) {
+          if (err) {
+            deferred.reject(err);
+          } else {
+            deferred.resolve(env);
+          }
+        });
+      }
```

Comment:
When I checked host.action, it always just seemed to return undefined; so I figured I need to use $q to promise-ify it. I want to bubble the result to the calling window so that the UI can actually take action based on if the save was successful or not.

I was originally going to have it so clicking on the save would start up a spinner in the save button, then on success the window would close. Though the styleguide has the spinner outside the button, which has issues -- so I took it out.

Is there some magic about return host.action() that I'm missing?